### PR TITLE
Database firestore

### LINKED ISSUE #24: Add firestore dependency

### PR DESCRIPTION
**Pull request by gruvw**
Waiting for #23 to merge before. Fixes #24.

Adds the Firestore dependency with a toy example collection that will be deleted in the future.

It is set up now with an easy way to test and mock the database of the project.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+  await Firebase.initializeApp(
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
+
+  runApp(
+    const ProviderScope(
+      child: ProximaApp(),
+    ),
+  );
 }
```
Comment (yoannLafore) [inline lib/main.dart:19]: Maybe we could add a global listener on the `userProvider` to listen to the authentication state and redirect toward the login screen if the user is disconnected.
This way, we can handle the case where the user is unexpectidly disconnected from an exterior source (for example if the user goes in his google account and select "disconnect from all devices").

**Review by yoannLafore — COMMENTED** [review]
Good job, thanks for your work !
Overall, I think that the code is well structured and nicely manages the login logic.

I have just found a few issues that I detail further in the review.

File: lib/utils/ui/circular_value.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+
+/// Utilitiy widget used to display a [CircularProgressIndicator] while waiting for an [AsyncValue] to complete; and another widget once the data resolves.
```
[the remark this answers is not in the document]
Comment (gruvw) [reply lib/utils/ui/circular_value.dart:4]: Applied in future commit :+1:

File: lib/services/login_service.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    // Obtain the auth details from the request
+    final GoogleSignInAuthentication? googleAuth =
+        await googleUser?.authentication;
+
+    // Create a new credential
+    final credential = GoogleAuthProvider.credential(
+      accessToken: googleAuth?.accessToken,
+      idToken: googleAuth?.idToken,
+    );
+
```
[the remark this answers is not in the document]
Comment (gruvw) [reply lib/services/login_service.dart:29]: This was resloved by https://github.com/ProximaEPFL/proxima/pull/23/commits/5c79d30a1dd1a718912eecc4d96f188e335f47e8 I pulled the changes into this branch.

File: lib/main.dart (hunk 1 above)
Comment (gruvw) [reply lib/main.dart:19]: Yes great suggestion! I added this provider onto the root widget of the application (`HomePage`) as it should never be popped from the widget stack, and we must attach this listener to a widget.

Check it out and let me know if you have further remarks.

**Review by yoannLafore — APPROVED** [review]
Thanks for the changes you made !
This looks good to me now.

**Comment by JoachimFavre** [comment]
I'm sorry, I don't have time to review this before the sprint meeting.

**Pull request merged**